framework/src/main/java/org/tron/core/db/Manager.java

```diff
@@ -1572,7 +1573,20 @@ private void startEventSubscribing() {
     }
   }
 
+  private void postSolidityTrigger(final long latestSolidifiedBlockNumber) {
+    if (eventPluginLoaded) {
+      SolidityTriggerCapsule solidityTriggerCapsule
+          = new SolidityTriggerCapsule(latestSolidifiedBlockNumber);
+      boolean result = triggerCapsuleQueue.offer(solidityTriggerCapsule);
+      if (!result) {
+        logger.info("too many trigger, lost solidified trigger, "
+            + "block number: {}", latestSolidifiedBlockNumber);
+      }
+    }
+  }
+
   private void postBlockTrigger(final BlockCapsule newBlock) {
+    postSolidityTrigger(getDynamicPropertiesStore().getLatestSolidifiedBlockNum());
```

Comment:
solidity trigger should not depends on block trigger. It could increase the probability of OOM

Reply:
ok, I update already.